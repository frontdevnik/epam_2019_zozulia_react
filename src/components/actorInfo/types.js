export const SHOW_ACTOR = 'SHOW_ACTOR';
export const SHOW_ACTOR_ERROR = 'SHOW_ACTOR_ERROR';
export const REMOVE_ACTOR = 'REMOVE_ACTOR';
