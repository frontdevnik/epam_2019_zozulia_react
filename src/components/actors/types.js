export const GET_ACTORS = 'GET_ACTORS';
export const ACTORS_ERROR = 'ACTORS_ERROR';
