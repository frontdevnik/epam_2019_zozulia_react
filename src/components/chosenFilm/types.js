export const CHOOSE_FILM = 'CHOOSE_FILM';
export const CHANGE_SELECTED_MOVIE_LIKES = 'CHANGE_SELECTED_MOVIE_LIKES';
export const CHANGE_SELECTED_MOVIE_STARS = 'CHANGE_SELECTED_MOVIE_STARS';
export const DELETE_MOVIE = 'DELETE_MOVIE';
export const SHOW_ACTOR = 'SHOW_ACTOR';
export const FETCHING_MOVIE = 'FETCHING_MOVIE';
export const LOADING_MOVIE = 'LOADING_MOVIE';
export const ERROR_MOVIE = 'ERROR_MOVIE';
export const REMOVE_MOVIE_REDUCER = 'REMOVE_MOVIE_REDUCER';
