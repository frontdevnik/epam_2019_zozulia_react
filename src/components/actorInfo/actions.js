import { SHOW_ACTOR, SHOW_ACTOR_ERROR, REMOVE_ACTOR } from './types';

const showActor = (payload) => ({
  type: SHOW_ACTOR,
  payload,
});

const showActorError = (payload) => ({
  type: SHOW_ACTOR_ERROR,
  payload,
});

export const fetchingActor = (id) => async (dispatch, _, api) => {
  try {
    const { data } = await api(`actors/${id}`);
    dispatch(showActor(data));
  } catch (error) {
    dispatch(showActorError(error));
  }
};

export const removeActor = () => ({
  type: REMOVE_ACTOR,
});
