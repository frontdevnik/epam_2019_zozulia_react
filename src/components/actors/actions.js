export const getActors = (state) => async (dispatch, _, api) => {
  const actors = api('actors');
  return { ...state, loading: false, actors };
};
