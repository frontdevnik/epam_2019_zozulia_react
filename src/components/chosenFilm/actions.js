import { SHOW_ACTOR, CHOOSE_FILM, ERROR_MOVIE, REMOVE_MOVIE_REDUCER } from './types';

export const showActorInfo = (payload) => ({
  type: SHOW_ACTOR,
  payload,
});

const getMovie = (payload) => ({
  type: CHOOSE_FILM,
  payload,
});

const errorMovie = (payload) => ({
  type: ERROR_MOVIE,
  payload,
});

export const fetchingMovie = (id) => async (dispatch, _, api) => {
  try {
    const { data } = await api(`movies/${id}`);
    dispatch(getMovie(data));
  } catch (error) {
    dispatch(errorMovie(error));
  }
};

export const removeMovieReducer = () => ({
  type: REMOVE_MOVIE_REDUCER,
});
