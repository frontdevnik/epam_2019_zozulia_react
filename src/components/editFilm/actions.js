import { EDIT_MOVIE } from './types';

export const editMovie = (payload) => ({
  type: EDIT_MOVIE,
  payload,
});
