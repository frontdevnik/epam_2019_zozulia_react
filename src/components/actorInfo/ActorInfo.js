import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchingActor, removeActor } from './actions';
import Loading from '../loading/Loading';
import Error from '../error/Error';
import ActorInfoUI from './ActorInfoUI';

function ActorInfo(props) {
  const { id } = useParams();
  const { fetchingActor, removeActor, actor, loading, error } = props;

  useEffect(() => {
    fetchingActor(id);

    return () => {
      removeActor();
    };
  }, [fetchingActor, removeActor, id]);

  if (loading && !actor) {
    return <Loading />;
  }

  if (error) {
    return <Error />;
  }

  return (
    <ActorInfoUI actor={actor} />
  );
}

ActorInfo.propTypes = {
  actor: PropTypes.object,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.oneOf([null]),
  ]),
  fetchingActor: PropTypes.func,
  removeActor: PropTypes.func,
};

const mapStateToProps = (state) => ({
  actor: state.actorReducer.actor,
  loading: state.actorReducer.loading,
  error: state.actorReducer.error,
});

const mapDispatchToProps = {
  fetchingActor,
  removeActor,
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default withConnect(ActorInfo);
