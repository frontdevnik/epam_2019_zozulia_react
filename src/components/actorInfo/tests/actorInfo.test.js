import React from 'react';
import enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ActorInfo from '../ActorInfo';
import * as actions from '../actions';
import * as types from '../types';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    match: {
      params: { id: 1 }
    }
  }),
}));

describe('<ActorInfo />', () => {
  let wrapper;
  let instance;
  let store;
  const initialState = {
    actorReducer: {
      actor: null,
      loading: true,
      error: null,
    },
  };
  const mockStore = configureStore();

  const props = {
    id: 3,
  }

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Provider store={store}><ActorInfo {...props} /></Provider>);
    instance = wrapper.instance();
  });

  it('should render loading', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render error', () => {
    store = mockStore({
      actorReducer: {
        actor: null,
        loading: false,
        error: true,
      },
    });
    wrapper = shallow(<Provider store={store}><ActorInfo {...props} /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render actor', () => {
    store = mockStore({
      actorReducer: {
        actor: {},
        loading: false,
        error: null,
      },
    });
    wrapper = shallow(<Provider store={store}><ActorInfo {...props} /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('actions', () => {
    it('should remove actor from store', () => {
      const expectedAction = {
        type: types.REMOVE_ACTOR,
      };
      expect(actions.removeActor()).toEqual(expectedAction);
    });
  });
});
