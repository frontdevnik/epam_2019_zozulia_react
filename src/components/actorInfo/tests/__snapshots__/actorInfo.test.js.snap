// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActorInfo /> should render actor 1`] = `"<div class=\\"aboutActor\\"><img class=\\"profileImg\\"/><div><p class=\\"actorInfo\\"><b>Name: </b></p><p class=\\"actorInfo\\"><b>Biography: </b></p></div></div>"`;

exports[`<ActorInfo /> should render error 1`] = `"<h1 style=\\"text-align:center\\">Sorry, the service is not available now, try visiting later</h1>"`;

exports[`<ActorInfo /> should render loading 1`] = `"<div class=\\"loader\\"><span>Loading...</span></div>"`;
